23/README.md

```diff
@@ -0,0 +1,283 @@
+
+```
+benchmark1@benchmark1:~$ sudo hdparm -Tt /dev/sda
+
+/dev/sda:
+ Timing cached reads:   19514 MB in  1.99 seconds = 9785.28 MB/sec
+ Timing buffered disk reads: 1484 MB in  3.00 seconds = 494.39 MB/sec
+```
+
+All machines were provided with the operating system *Ubuntu 18.04.1 LTS (Bionic Beaver)*. For simplicity we provided all machines (both the machine generating the workload and the machines running the BigchainDB nodes) with the same software:
+
+- BigchainDB, [commit 6a9064196ad49baf3933051be6e116a3440fbad2][bdb:git-commit]
+- Tendermint 0.22.8
+- MongoDB v3.6.3
+
+This procedure was automated by a [custom script][self:provide.sh].
+
+Three main configuration files were generated and edited (all of them are included as extra files in this BEP):
+- The Tendermint [`config.toml`](self:config.toml) configuration file (default location: `${HOME}/.tendermint/config/config.toml`) was generated via `tendermint init` and customized.
+- The Tendermint [`genesis.json`][self:genesis.json] file (generated though the previous command `tendermint init`) was customized with the public keys of the four nodes. It was then copied into all machines under `${HOME}/.tendermint/config/genesis.json`.
+- The BigchainDB [`.bigchaindb`][self:bigchaindb.json] was generated via `bigchaindb configure` and copied into all machines under `${HOME}/.bigchaindb`.
+
+We also had a **setup** and **teardown** procedure we run before and after every test.
+
+The **setup** procedure was responsible to start the BigchainDB and the Tendermint process.
+```bash
+#!/bin/bash
+
+nohup bigchaindb start > ${HOME}/bigchaindb.log 2>&1 &
+nohup tendermint node > ${HOME}/tendermint.log 2>&1 &
+```
+
+The **teardown** procedure was responsible to stop the BigchainDB and the Tendermint process, and to reset the state on MongoDB and on the Tendermint LevelDB.
+```bash
+#!/bin/bash
+
+pkill tendermint
+pkill bigchaindb
+bigchaindb -y drop
+tendermint unsafe_reset_all
+```
+
+### Experiments
+Three experiments were performed.
+
+The first one was run using BigchainDB without any special option.
+
+The second one used the experimental feature for parallel validation, enabled on start with `bigchaindb start --experimental-parallel-validation`.
+
+The third one used the experimental feature for parallel validation, enabled on start with `bigchaindb start --experimental-parallel-validation`, and was run with a smaller amount of requests to test the peak performance of the system.
+
+### Experiment 1: finalize 1,000,000 transactions
+For each node in the network, we started all services using the startup script. After some seconds, we switched to the workload-generating virtual machine, and ran:
+
+```
+bigchaindb-benchmark\
+    --csv results/master-32proc-1000000sync.csv\
+    --processes=32\
+    --peer http://10.240.0.4:9984\
+    --peer http://10.240.0.5:9984\
+    --peer http://10.240.0.6:9984\
+    --peer http://10.240.0.7:9984\
```

Comment:
Should we comment that if they try to repeat our experiments, then their IP addresses will almost certainly be different?

Reply:
Yes, I'm reporting what we used. Don't want to sound rude, but if they don't understand this, I'm not sure they can actually do a benchmark. But if we want to add a comment about that, we should write it at the beginning of the document, because it applies also to the configuration files presented in this BEP.

Reply:
Let's add a comment _somewhere_ about how, if they want to reproduce our results, their node IP addresses will be different.

Reply:
What about adding at the beginning of the **Methods** section the following sentence:
> Note: to reproduce our results, make sure to adapt the configuration files, scripts, and commands to your setup.

Reply:
That sounds quite open-ended and scary. Maybe you could make it less scary by writing a specific example of something they will have to change:

> Note: to reproduce our results, make sure to adapt the configuration files, scripts, and commands to your setup. For example, your nodes will have different IP addresses.

Reply:
It sounds much better! :smile_cat: